src/pkg/utils.go

```diff
@@ -140,3 +146,32 @@ func IsValidTime(t time.Time) bool {
 	// We could be even more conservative and check for > 2000 or so, but this is more predictable.
 	return t.Year() > 1970
 }
+
+func Compare(actual []byte, goldenFile string) error {
+	// Replace the absolute path in context to make the golden file portable
+	absPath, _ := filepath.Abs(goldenFile)
+	actual = bytes.ReplaceAll(actual, []byte(filepath.Dir(absPath)), []byte{'.'})
+
+	golden, err := os.ReadFile(goldenFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("Failed to read golden file: %w", err)
+		}
+		return os.WriteFile(goldenFile, actual, 0644)
+	} else {
+		if err := Diff(string(actual), string(golden)); err != nil {
+			return fmt.Errorf("%s %w", goldenFile, err)
+		}
+	}
+	return nil
+}
+
+func Diff(actualRaw, goldenRaw string) error {
```

Comment:
Can still be private:
```suggestion
		if err := diff(string(actual), string(golden)); err != nil {
			return fmt.Errorf("%s %w", goldenFile, err)
		}
	}
	return nil
}

func diff(actualRaw, goldenRaw string) error {
```

Reply:
I cannot access this from package mcp if it is private.